<?xml version="1.0"?>
<sdf version="1.7">
  <model name="box_usv">
    <link name="base_link">
      <pose>0 0 0 0 0 0</pose>
      <inertial>
        <pose>0.0012 0 -0.05 0 0 0</pose>
        <mass>100</mass>
        <inertia>
          <ixx>1.9833</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>2.358</iyy>
          <iyz>0</iyz>
          <izz>3.292</izz>
        </inertia>
      </inertial>
      <visual name="hull_visual">
        <geometry>
          <box><size>1 0.6 0.23</size></box>
        </geometry>
      </visual>
      <collision name="hull_collision">
        <geometry>
          <box><size>1 0.6 0.23</size></box>
        </geometry>
      </collision>
    </link>

    <link name="camera_link">
      <pose>0 0 0.6 0 0 0</pose>
      <sensor name="camera_sensor" type="camera">
        <camera>
          <horizontal_fov>1.047</horizontal_fov>
          <image>
            <width>640</width>
            <height>480</height>
            <format>R8G8B8</format>
          </image>
          <clip>
            <near>0.1</near>
            <far>100</far>
          </clip>
        </camera>
        <always_on>true</always_on>
        <update_rate>30</update_rate>
        <visualize>true</visualize>
      </sensor>
    </link>
    <joint name="camera_joint" type="fixed">
      <parent>base_link</parent>
      <child>camera_link</child>
    </joint>

    <link name="thruster_L">
      <pose>-0.525 -0.15 0 0 1.57 0</pose>
      <visual name="visual">
        <geometry><box><size>0.1 0.07 0.05</size></box></geometry>
      </visual>
      <collision name="collision">
        <geometry><box><size>0.1 0.07 0.05</size></box></geometry>
      </collision>
      <inertial>
        <mass>0.525</mass>
        <inertia>
          <ixx>0.00032375</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0.0018594</iyy>
          <iyz>0</iyz>
          <izz>0.0019644</izz>
        </inertia>
      </inertial>
    </link>
    <joint name="joint_propeller_port_L" type="revolute">
      <parent>base_link</parent>
      <child>thruster_L</child>
      <axis><xyz>0 0 1</xyz></axis>
    </joint>

    <link name="thruster_R">
      <pose>-0.525 0.15 0 0 1.57 0</pose>
      <visual name="visual">
        <geometry><box><size>0.1 0.07 0.05</size></box></geometry>
      </visual>
      <collision name="collision">
        <geometry><box><size>0.1 0.07 0.05</size></box></geometry>
      </collision>
      <inertial>
        <mass>0.525</mass>
        <inertia>
          <ixx>0.00032375</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0.0018594</iyy>
          <iyz>0</iyz>
          <izz>0.0019644</izz>
        </inertia>
      </inertial>
    </link>
    <joint name="joint_propeller_port_R" type="revolute">
      <parent>base_link</parent>
      <child>thruster_R</child>
      <axis><xyz>0 0 1</xyz></axis>
    </joint>

    <plugin filename="ignition-gazebo-hydrodynamics-system" name="ignition::gazebo::systems::Hydrodynamics">
      <link_name>base_link</link_name>
      <xDotU>-0.04876161</xDotU>
      <yDotV>-1.26324739</yDotV>
      <zDotW>-1.26324739</zDotW>
      <kDotP>0</kDotP>
      <mDotQ>-0.3346</mDotQ>
      <nDotR>-0.3346</nDotR>
      <xUabsU>-0.62282</xUabsU>
      <xU>-5</xU>
      <yVabsV>-60.127</yVabsV>
      <yV>-5</yV>
      <zWabsW>-6.0127</zWabsW>
      <zW>-100</zW>
      <kPabsP>-0.001916</kPabsP>
      <kP>-1</kP>
      <mQabsQ>-6.32698957</mQabsQ>
      <mQ>-1</mQ>
      <nRabsR>-6.32698957</nRabsR>
      <nR>-1</nR>
    </plugin>
    <plugin filename="ignition-gazebo-thruster-system" name="ignition::gazebo::systems::Thruster">
      <joint_name>joint_propeller_port_L</joint_name>
      <namespace>box_usv</namespace>
      <topic>thruster_L</topic>
      <thrust_coefficient>0.001</thrust_coefficient>
      <fluid_density>1000</fluid_density>
      <propeller_diameter>0.4</propeller_diameter>
    </plugin>
    <plugin filename="ignition-gazebo-thruster-system" name="ignition::gazebo::systems::Thruster">
      <joint_name>joint_propeller_port_R</joint_name>
      <namespace>box_usv</namespace>
      <topic>thruster_R</topic>
      <thrust_coefficient>0.001</thrust_coefficient>
      <fluid_density>1000</fluid_density>
      <propeller_diameter>0.4</propeller_diameter>
    </plugin>
    <plugin filename="ignition-gazebo-sensors-system" name="ignition::gazebo::systems::Sensors">
      <render_engine>ogre2</render_engine>
    </plugin>
  </model>
</sdf>
